### PR TITLE
Add TemplateProject for generating cookiecutter

### PR DESCRIPTION
**Pull request by sharplet**
It's difficult to work on `{{ cookiecutter.project_name }}.xcodeproj` as it doesn't parse correctly as a project file. This change adds `TemplateProject`, which is editable as a normal Xcode project file, and a `bin/generate` script to generate the cookiecutter from it.

I haven't tried this, but seems like it should be possible to take this one step further, and dynamically generate the cookiecutter in `hooks/pre_gen_project.sh`. We could then add `{{ cookiecutter.project_name }}/` to `.gitignore`, which would remove the duplicated changes in commits.

### CODE REVIEW

File: TemplateProject/Cartfile.resolved [hunk 1]
```diff
@@ -0,0 +1,6 @@
+github "steipete/Aspects" "1.4.2"
```
Comment (tonyd256) [inline TemplateProject/Cartfile.resolved]: does this resolve file match the Cartfile?

Comment (sharplet) [reply TemplateProject/Cartfile.resolved]: This one's a transitive dep of Perform

File: TemplateProject/Cartfile [hunk 2]
```diff
@@ -0,0 +1 @@
+github "thoughtbot/Argo" "master"
```
Comment (tonyd256) [inline TemplateProject/Cartfile:1]: this can probably come off master now

File: TemplateProject/Cartfile.resolved (hunk 1 above)
Comment (sharplet) [reply TemplateProject/Cartfile.resolved]: Also...it shouldn't be checked in, woops

File: TemplateProject/TemplateProject.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -0,0 +1,645 @@
...
+		728E193911D9B3956F2730E0 /* Info.plist */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
+		74AF5450F820AF3F3A02545D /* AppDelegate.swift */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = sourcecode.swift; path = AppDelegate.swift; sourceTree = "<group>"; };
+		8491BE79156DCEE4E36BB513 /* LaunchScreen.xib */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = file.xib; path = LaunchScreen.xib; sourceTree = "<group>"; };
+		C7FF830DA698871C1495778E /* Images.xcassets */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = folder.assetcatalog; path = Images.xcassets; sourceTree = "<group>"; };
+		CBCE1D85EA9B3F07566B6E71 /* Tests.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = Tests.xctest; sourceTree = BUILT_PRODUCTS_DIR; };
+		F83205321D83694700047B79 /* Argo.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Argo.framework; path = Carthage/Build/iOS/Argo.framework; sourceTree = "<group>"; };
```
Comment (tonyd256) [inline TemplateProject/TemplateProject.xcodeproj/project.pbxproj:57]: I don't think this is the way frameworks should look in the file references. Perhaps we could double check a working project?

File: TemplateProject/TemplateProject.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -0,0 +1,645 @@
...
+		F87108221D8706D60003294E /* Runes.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Runes.framework; path = Carthage/Build/iOS/Runes.framework; sourceTree = "<group>"; };
+		F8C3823B1D832B7B00FF4F67 /* Nimble.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Nimble.framework; path = Carthage/Build/iOS/Nimble.framework; sourceTree = "<group>"; };
+		F8C3823C1D832B7B00FF4F67 /* Quick.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = Quick.framework; path = Carthage/Build/iOS/Quick.framework; sourceTree = "<group>"; };
```
Comment (tonyd256) [inline TemplateProject/TemplateProject.xcodeproj/project.pbxproj:60]: There seems to be a lot of frameworks missing that are included in Cartfile.resolved

File: TemplateProject/Cartfile.resolved (hunk 1 above)
Comment (tonyd256) [reply TemplateProject/Cartfile.resolved]: not just this one but the Cartfile only contains Argo ... but there is a lot in this file

File: TemplateProject/TemplateProject.xcodeproj/project.pbxproj (hunk 4 above)
Comment (sharplet) [reply TemplateProject/TemplateProject.xcodeproj/project.pbxproj:60]: You're not wrong

File: TemplateProject/TemplateProject.xcodeproj/project.pbxproj (hunk 3 above)
Comment (sharplet) [reply TemplateProject/TemplateProject.xcodeproj/project.pbxproj:57]: Doesn't seem wrong to me?

File: TemplateProject/Cartfile (hunk 2 above)
Comment (sharplet) [reply TemplateProject/Cartfile:1]: #16

File: TemplateProject/Cartfile.resolved (hunk 1 above)
Comment (sharplet) [reply TemplateProject/Cartfile.resolved]: Oh right, Perform isn't in master yet. Weird. I'll fix it

File: TemplateProject/TemplateProject.xcodeproj/project.pbxproj (hunk 4 above)
Comment (sharplet) [reply TemplateProject/TemplateProject.xcodeproj/project.pbxproj:60]: Ok so Argo (and thus Runes) are in Cartfile, and Quick/Nimble are in Cartfile.private. Should be good now.

**Comment by sharplet** [comment]
> I haven't tried this, but seems like it should be possible to take this one step further, and dynamically generate the cookiecutter in `hooks/pre_gen_project.sh`.

Ok so cookiecutter doesn't like this.

**Pull request closed without merging**